surf-bitmap-provider-generator/src/main/kotlin/dev/slne/surf/bitmap/generator/command/subcommands/GenerateCommand.kt

```diff
@@ -0,0 +1,180 @@
+package dev.slne.surf.bitmap.generator.command.subcommands
+
+import com.github.shynixn.mccoroutine.folia.launch
+import dev.jorel.commandapi.CommandAPICommand
+import dev.jorel.commandapi.kotlindsl.*
+import dev.slne.surf.bitmap.generator.generator.BitmapProviderGenerator
+import dev.slne.surf.bitmap.generator.generator.GeneratorResult
+import dev.slne.surf.bitmap.generator.plugin
+import dev.slne.surf.bitmap.generator.utils.PermissionRegistry
+import dev.slne.surf.surfapi.core.api.messages.adventure.sendText
+import org.bukkit.command.CommandSender
+import kotlin.io.path.div
+import kotlin.time.measureTime
+
+fun CommandAPICommand.generateCommand() = subcommand("generate") {
+    withPermission(PermissionRegistry.LETTERGEN_COMMAND_GENERATE)
+
+    textArgument("name")
+    textArgument("foregroundHex")
+    textArgument("backgroundHex")
+    textArgument("configPath")
+    textArgument("texturePath")
+    booleanArgument("force", true)
+
+    anyExecutor { sender, args ->
+        val name: String by args
+        val foregroundHex: String by args
+        val backgroundHex: String by args
+        val configPath: String by args
+        val texturePath: String by args
+        val force: Boolean? by args
+        val override = force ?: false
+
+        if (name.length !in 3..64) {
+            sender.sendText {
+                appendPrefix()
+
+                error("Der Name muss zwischen 3 und 64 Zeichen lang sein.")
+            }
+            return@anyExecutor
+        }
+
+        val realForegroundHex = foregroundHex.replace("#", "")
+        val realBackgroundHex = backgroundHex.replace("#", "")
+
+        if (realForegroundHex.length != 6 || realBackgroundHex.length != 6) {
+            sender.sendText {
+                appendPrefix()
+
+                error("Die Hex-Farben müssen 6 Zeichen (7 Zeichen mit #) lang sein.")
+            }
+
+            return@anyExecutor
+        }
+
+        if (!checkValidHex(realForegroundHex) || !checkValidHex(realBackgroundHex)) {
+            sender.sendText {
+                appendPrefix()
+
+                error("Die Hex-Farben müssen aus Zahlen und Buchstaben von a bis f bestehen.")
+            }
+
+            return@anyExecutor
+        }
+
+        if (configPath.isBlank()) {
+            sender.sendText {
+                appendPrefix()
+
+                error("Der Pfad muss angegeben werden.")
+            }
+
+            return@anyExecutor
+        }
+
+        if (texturePath.isBlank()) {
+            sender.sendText {
+                appendPrefix()
+
+                error("Der Pfad muss angegeben werden.")
+            }
+
+            return@anyExecutor
+        }
+
+        val nexoPluginFolderPath = plugin.dataPath.parent / "Nexo"
```

Comment:
[Server.html#getPluginsFolder()](https://jd.papermc.io/paper/1.21.5/org/bukkit/Server.html#getPluginsFolder())

```suggestion
        val nexoPluginFolderPath = server.pluginsFolder.toPath() / "Nexo"
```